src/sec_certs/br1/chapter_parsing/mapper.py

```diff
@@ -0,0 +1,82 @@
+import re
+from pathlib import Path
+
+import regex
+
+import sec_certs.br1.config.constants as config
+from sec_certs.br1.models.chapter import Chapter
+
+from .chapter_utils import chapters_from_json, traverse_chapters
+
+
+def substitute(title: str) -> str:
+    """Remove spaces and dashes."""
+    return re.sub(r"[ \-–—‒]", "", title)
+
+
+def build_chapter_regex(chapters: list[Chapter], chapter_num: int, subchapter_num: int) -> str:
+    """Construct a fuzzy regex for chapter titles."""
+    base_chapter = chapters[chapter_num - 1]
+    title = base_chapter.title
+    if subchapter_num > 0:
+        title = base_chapter.subchapters[subchapter_num - 1].title
+
+    title = substitute(title)
+
+    return rf"^(##|Section)*{chapter_num}(\.?{subchapter_num})?\.?{title}$"
+
+
+# Core extraction logic
+def extract_chapters_from_text(text: str) -> list[Chapter]:
```

Comment:
As I understand it, this uses the Markdown output of Docling to parse out the chapter. Thus, it introduces additional dependencies on Markdown parsing and does a somewhat unnecessary round trip where Docling parses the PDF to some internal format (which it dumps as JSON) and then serializes it to Markdown which is parsed here. I think this is an OK solution for now. But when #525 lands, we should consider parsing out the chapters directly from that JSON.